include/eve/detail/function/simd/x86/swizzle.hpp

```diff
@@ -0,0 +1,258 @@
+          constexpr auto lp = pattern_view<0,4,8>(q);
+          constexpr auto hp = pattern_view<4,8,8>(q);
+
+                if constexpr( lp < 4 && hp >= 4) return process_zeros(that_t{v[lp],v[hp]},q);
+          else  if constexpr( hp < 4 && lp >= 4) return process_zeros(that_t{v[lp],v[hp]},q);
+          else  return basic_swizzle_(EVE_RETARGET(cpu_),v,q);
+        }
+      }
+      else
+      {
+        return basic_swizzle_(EVE_RETARGET(cpu_),v,q);
+      }
+    }
+  }
+
+  //================================================================================================
+  // AVX+ variant
+  //================================================================================================
+  template<typename T, typename N, x86_abi ABI, shuffle_pattern Pattern>
+  EVE_FORCEINLINE auto basic_swizzle_( EVE_SUPPORTS(avx_), wide<T,N,ABI> const& v, Pattern const& p)
+  {
+    constexpr auto cd = N::value;
+    constexpr auto sz = Pattern::size(cd);
+    using that_t      = as_wide_t<wide<T,N,ABI>,fixed<sz>>;
+
+    constexpr auto width_in   = cd*sizeof(T);
+    constexpr auto width_out  = sz*sizeof(T);
+
+    constexpr Pattern q = {};
+
+    // We're swizzling so much we aggregate the output
+    if constexpr( has_aggregated_abi_v<that_t> )
+    {
+      return aggregate_swizzle(v,q);
+    }
+    else if constexpr( width_in == 64 )
+    {
+      /// TODO: AVX512VBMI supports 128-bits permutexvar
+      if constexpr(width_out <= 16)       return basic_swizzle_(EVE_RETARGET(cpu_),v,q);
+      else if constexpr(width_out == 32)
+      {
+              if constexpr( q <  cd/2 )  return v.slice(lower_)[ q ];
+        else  if constexpr( q >= cd/2 )  return v.slice(upper_)[ slide_pattern<cd/2,sz>(q) ];
+        /// TODO: optimize using SSSE3 + binary shuffle
+        else                             return basic_swizzle_(EVE_RETARGET(cpu_),v,q);
+      }
+      else
+      {
+        constexpr auto c = categorize<that_t>();
+        auto const     m = as_indexes<that_t>(q);
+        that_t s;
+
+              if constexpr(c == category::float64x8 )                       s = _mm512_permutexvar_pd(m,v);
+        else  if constexpr(c == category::float32x16)                       s = _mm512_permutexvar_ps(m,v);
+        else  if constexpr(match(c,category::int32x16,category::uint32x16)) s = _mm512_permutexvar_epi32(m,v);
+        else  if constexpr(match(c,category::int16x32,category::uint16x32)) s = _mm512_permutexvar_epi16(m,v);
+        else  return basic_swizzle_(EVE_RETARGET(cpu_),v,q);
+
+        return process_zeros(s,q);
+      }
+    }
+    else if constexpr( width_in == 32 )
+    {
+      if constexpr(width_out <= 16)
+      {
+              if constexpr( q <  (cd/2) ) return v.slice(lower_)[ q ];
+        else  if constexpr( q >= (cd/2) ) return v.slice(upper_)[ slide_pattern<cd/2,sz>(q) ];
+        /// TODO: optimize using SSSE3 + binary shuffle
+        else                              return basic_swizzle_(EVE_RETARGET(cpu_),v,q);
```

Comment:
Again retarget cpu and not sse2

Reply:
because those are AVX register, you can not pass them to sse2